js/ui/soundable.js

```diff
@@ -0,0 +1,73 @@
+const Lang = imports.lang;
+const St = imports.gi.St;
+const GObject = imports.gi.GObject;
+const SoundServer = imports.misc.soundServer;
+
+
+var Button = new Lang.Class({
+    Name: 'SoundableButton',
+    Extends: St.Button,
+    Properties: {
+        'click-sound-event-id': GObject.ParamSpec.string(
+            'click-sound-event-id',
+            '',
+            '',
+            GObject.ParamFlags.READWRITE,
+            ''
+        ),
+        'hover-sound-event-id': GObject.ParamSpec.string(
+            'hover-sound-event-id',
+            '',
+            '',
+            GObject.ParamFlags.READWRITE,
+            ''
+        ),
+        'stop-hover-sound-on-click': GObject.ParamSpec.boolean(
+            'stop-hover-sound-on-click',
+            '',
+            '',
+            GObject.ParamFlags.READWRITE,
+            ''
+        )
+    },
+
+    _init: function(params) {
+        this.parent(params);
+
+        if (this.stop_hover_sound_on_click === undefined)
+            this.stop_hover_sound_on_click = false;
```

Comment:
If we make the `stop-hover-on-sound-click` property a construct one (by setting `GObject.ParamFlag.CONSTRUCT`), you should not need this, as the setter will be called during initialization.